extide/gradle/netbeans-gradle-tooling/src/main/java/org/netbeans/modules/gradle/tooling/NbProjectInfoBuilder.java

```diff
@@ -290,7 +290,8 @@ private void detectTaskProperties(NbProjectInfoModel model) {
         Map<String, Object> taskProperties = new HashMap<>();
         Map<String, String> taskPropertyTypes = new HashMap<>();
         
-        for (Task task : project.getTasks().getAsMap().values()) {
+        // make a copy of the task map; may mutate.
+        for (Task task : new ArrayList<>(project.getTasks().getAsMap().values())) {
```

Comment:
what happens when map mutates while ArrayList constructor rxecutes

Reply:
This part of build does not run concurrently AFAIK,  the issue is that the set of tasks mutates during task inspection, inside the cycle